File: backend/app/schemas/finding.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class FindingResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    analysis_run_id: UUID
    rule_id: str
    severity: str
    title: str
    description: str | None
    file_path: str | None
    line_start: int | None
    line_end: int | None
    status: str
    created_at: datetime

File: frontend/src/types/finding.ts
export type Severity =
  | 'critical'
  | 'high'
  | 'medium'
  | 'low'
  | 'info';

export type VerificationStatus =
  | 'verified'
  | 'unverified'
  | 'partial';

export interface Finding {
  id: string;
  analysis_run_id: string;
  rule_id: string;
  severity: Severity;
  title: string;
  description: string | null;
  file_path: string | null;
  line_start: number | null;
  line_end: number | null;
  status: string;
  created_at: string;
}

export type FindingFilter =
  | 'all'
  | 'critical'
  | 'high'
  | 'medium'
  | 'low';